design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,271 @@
+        3. Instructions to configure Hive Metastore with lakeFS.
+        4. Generated this README file and committed it.
+    ```
+    and upload and commit it to lakeFS.
+
+### Happy flow - Import only
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket (after they validated that the lakeFS role has the right permissions and that the bucket has the correct policy), and clicks ‘OK’.
+A progress bar or a counter will show the progress of the import process and will signal once it’s over.
+5. The wizard asks the user: “How are you using Spark”, and offers three alternatives: Databricks, AWS EMR, and spark-submit. The user skips this step using the skip button.
+6. The wizard will generate a README file as follows:
+   ```markdown
+   1. Created a repository "spark-repo" and branch "main".
+   2. Imported data from <S3 location>
+   3. Generated this README file and committed it.
+   ```
+   and upload and commit it to lakeFS.
+
+### Sad flow - No import permissions
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket, and clicks ‘OK’.
+   1. The import functionality panics as there are no permissions to access the given storage.
+   2. The wizard will show an error message like: “Please verify your lakeFS server and storage have the required permissions” and link to the docs to show the needed permissions.
+5. Continue as above 
+- The generated README will not include the import step.
+
+### Sad flow - No credentials' generation permissions
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user skips this step using the skip button.
+5. The wizard asks the user: “How are you using Spark”, and offers three alternatives: Databricks, AWS EMR, and spark-submit. The user chooses ‘Databricks’. It also asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL). The user types the endpoint.
+6. The wizard sends a GET request to the templating service with a query string of the format:
```

Comment:
Optionally, we might use the templating service as the source for the "how are you using this" data.  It's some JSON, that needs to be generated somewhere, and may have some fields to specify.  Looks like another use-case for templates.